src/famfs_lib.c

```diff
@@ -236,24 +236,78 @@ famfs_print_uuid(const uuid_le *uuid)
 	printf("%s\n", uuid_str);
 }
 
-#define SYS_UUID_PATH "/sys/devices/virtual/dmi/id/product_uuid"
+#define SYS_UUID_FILE "/opt/famfs/system_uuid"
+#define SYS_UUID_DIR "/opt/famfs"
+
+/*
+ * Check if uuid file exists, if not, create it
+ * and update it with a new uuid.
+ *
+ */
+int famfs_create_sys_uuid_file()
+{
+	int uuid_fd, rc;
+	char uuid_str[37];  /* UUIDs are 36 char long, plus null terminator */
+	uuid_t local_uuid;
+	struct stat st = {0};
+	uuid_le sys_uuid;
+
+	/* Do nothing if file is present */
+	if (stat(SYS_UUID_FILE, &st) == 0)
+		return 0;
+
+	/* File not found, check for directory */
+	rc = stat(SYS_UUID_DIR, &st);
+	if (rc < 0 && errno == ENOENT) {
+		/* No directory found, create one */
+		rc = mkdir(SYS_UUID_DIR, 0644);
+		if (rc) {
+			fprintf(stderr, "%s: dir create error %s, errno: %d\n",
+				__func__, SYS_UUID_DIR, errno);
+			return -1;
+		}
+	}
+
+	uuid_fd = open(SYS_UUID_FILE, O_RDWR|O_CREAT, 0444);
+	if (uuid_fd < 0) {
+		fprintf(stderr, "%s: failed to open/create %s errno %d.\n",
+			__func__, SYS_UUID_FILE, errno);
+		return -1;
+	}
+	famfs_uuidgen(&sys_uuid);
+	memcpy(&local_uuid, &sys_uuid, sizeof(sys_uuid));
+	uuid_unparse(local_uuid, uuid_str);
+	rc = write(uuid_fd, uuid_str, 37);
+	if (rc < 0) {
```

Comment:
If the write fails, I think there will be an invalid uuid file left behind. Probably need to unlink it.

Reply:
Agreed, good catch, unlinked.